### PR TITLE
deb: add ubuntu 22.10 kinetic

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/docker-ce-packaging/pull/779

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: pkg/docker-engine/Dockerfile [hunk 1]
```diff
@@ -96,7 +96,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends apt-utils bash
 RUN <<EOT
   set -e
   case "$PKG_RELEASE" in
-    ubuntu2004|ubuntu2204)
+    ubuntu2004|ubuntu2204|ubuntu2210)
       if [ "$(dpkg-divert --truename /usr/bin/man)" = "/usr/bin/man.REAL" ]; then
```
Comment (thaJeztah) [inline pkg/docker-engine/Dockerfile:100]: I guess the only exception now is 18.04? Anything newer than that is now shipping without man-pages as default.

Wondering if we should make a reverse rule (less maintenance), and add a TODO to remove it once 18.04 is EOL, which looks to be April next year; https://github.com/docker/docker-ce-packaging/blob/396301be059a83162d39f5cc569998e7656492ce/Jenkinsfile#L16

File: pkg/docker-cli/Dockerfile [hunk 2]
```diff
@@ -94,7 +94,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends apt-utils bash
 RUN <<EOT
   set -e
   case "$PKG_RELEASE" in
-    ubuntu2004|ubuntu2204)
+    ubuntu2004|ubuntu2204|ubuntu2210)
```
Comment (thaJeztah) [inline pkg/docker-cli/Dockerfile:97]: (same for this one)

File: common/packages.hcl [hunk 3]
```diff
@@ -128,6 +128,16 @@ target "_pkg-ubuntu2204" {
   }
 }
 
+target "_pkg-ubuntu2210" {
+  args = {
+    PKG_RELEASE = "ubuntu2210"
+    PKG_TYPE = "deb"
+    PKG_DISTRO = "ubuntu"
+    PKG_SUITE = "kinetic"
+    PKG_BASE_IMAGE = "ubuntu:kinetic"
```
Comment (thaJeztah) [inline common/packages.hcl:137]: I always have to look up the code-names; even wondering if we should start using the CalVer versions for the image references instead (`ubuntu:22.10`). Either way; not for this PR (if we do, we should do it for all of them)

**Review by thaJeztah — APPROVED** [review]
LGTM, but left a suggestion for the man-pages block (okay for a follow-up)

File: pkg/docker-engine/Dockerfile (hunk 1 above)
Comment (crazy-max) [reply pkg/docker-engine/Dockerfile:100]: Ah right that makes sense

File: common/packages.hcl (hunk 3 above)
Comment (crazy-max) [reply common/packages.hcl:137]: Agree yes, I just followed `docker-ce-packaging` scheme.

Comment (thaJeztah) [reply common/packages.hcl:137]: Yes, we do a bit of a mix between "codename" and "version" (packages are named after the codename). I usually know the names for the LTS releases, but always forget the non-LTS ones 😂

**Pull request merged**